firmware/integration/rusefi_config.txt

```diff
@@ -1486,7 +1485,40 @@ tChargeMode_e tChargeMode;
 int8_t[MAX_CYLINDER_COUNT iterate] fuelTrim;;"Percent", @@PERCENT_TRIM_BYTE_PACKING_DIV@@, 0, -25, 25, 2
 
 	float[GAP_TRACKING_LENGTH iterate] triggerGapOverrideTo;;"to", 1, 0, 0, 20, 2
-	int[326] mainUnusedEnd;;"units", 1, 0, -20, 100, 0
+
+! Someday there will be a 6th option for BMW S55 that uses a separate shaft just for HPFP
+#define hpfp_cam_e_enum "NONE", "Intake 1", "Exhaust 1", "Intake 2", "Exhaust 2"
```

Comment:
separate shaft should just be an "unphased" option, right? There are other engines out there with variable cams, with the pump non-variable (for example, the C8 Z06's LT6 engine :) )

Reply:
or is that what `none` means?

Reply:
BMW N55/S55 engines actually have a separate shaft with a separate cam sensor.  Maybe it's so they don't have to time the pump shaft TDC on engine rebuilds?

https://www.realoem.com/bmw/enUS/showparts?id=3A93-USA-05-2015-F30-BMW-335i&diagId=11_4779
That shows where the HPFP mounts, along with part 6 - cam sensor.

NONE means no VVT.  Not sure those engines exist, but hey, why not....

Reply:
Was looking through Motec documentation, and they have a setting for "reduction" meaning the pump shaft isn't 1:1 with the crank.  I bet that's what the extra cam sensor is for.  Since the pump valves themselves can float if driven too fast, then having a reduction gear on the pump shaft lets you rev the engine higher.  It lets you effectively get 2.8 etc lobes per cam instead of having to choose between 2 and 3.